src/lib.rs

```diff
@@ -144,43 +144,49 @@ pub fn rebuild(stack: &mut Vec<Node>, proof: &Multiproof) -> Node {
                     children[*digit] = node;
                     stack.push(FullNode(children))
                 } else {
-                    panic!("Could not pop a value from the stack, that is required for a BRANCH")
+                    return Err(format!(
+                        "Could not pop a value from the stack, that is required for a BRANCH"
+                    ));
                 }
             }
             EXTENSION(key) => {
                 if let Some(node) = stack.pop() {
                     stack.push(Extension(key.to_vec(), Box::new(node)));
                 } else {
-                    panic!("Could not find a node on the stack, that is required for an EXTENSION")
+                    return Err(format!(
+                        "Could not find a node on the stack, that is required for an EXTENSION"
+                    ));
```

Comment:
If you're using `format!`, let's also add the key so that the user has an idea of which key went wrong

Reply:
Good point, updated